middleware/grpc_middleware_test.go

```diff
@@ -378,3 +376,57 @@ func TestMiddleware_EnvironmentIsolation(t *testing.T) {
 		})
 	}
 }
+
+func TestMiddleware_LoadRulesFromJSON(t *testing.T) {
+	ctx := context.Background()
+
+	rulesFile := writeTestConfig(t, "access-rules-*.json", `
+[{
+  "authenticators": [
+   {
+    "handler": "noop"
+   }
+  ],
+  "authorizer": {
+   "handler": "allow"
+  },
+  "id": "some-rule-id",
+  "match": {
+   "methods": [
+    "POST"
+   ],
+   "url": "<(https|http)>://example.com:8080/service/webhooks<(|/.*)>"
+  },
+  "mutators": [
+   {
+    "handler": "noop"
+   }
+  ],
+  "upstream": {
+   "preserve_host": true,
+   "strip_path": "/service",
+   "url": "http://example.svc.cluster.local"
+  }
+}]
+`)
+
+	configFile := writeTestConfig(t, "config-*.yaml", fmt.Sprintf(`
+access_rules:
+  matching_strategy: regexp
+  repositories:
+  - file://%s
+`, rulesFile))
+
+	regPtr := new(driver.Registry)
+	_, err := middleware.New(ctx,
+		middleware.WithConfigFile(configFile),
+		middleware.WithRegistry(regPtr),
+	)
+	require.NoError(t, err)
+	reg := *regPtr
+
+	time.Sleep(100 * time.Millisecond)
+
+	_, err = reg.RuleRepository().Get(ctx, "some-rule-id")
+	require.NoError(t, err)
```

Comment:
Maybe add a small assertion that verifies the content was loaded?

Reply:
It already verifies it through the id lookup (which failed before), but I can add some more content tests.